interop-testing/src/test/java/io/grpc/testing/integration/NettyFlowControlTest.java

```diff
@@ -0,0 +1,374 @@
+      testsFailed++;
+    }
+  }
+
+  /**
+   * Resets client/server and their flow control windows
+   *
+   * @param serverFlowControlWindow
+   * @param clientFlowControlWindow
+   */
+  private void resetConnection(int serverFlowControlWindow, int clientFlowControlWindow) {
+    if (channel != null) {
+      if (!channel.isShutdown()) {
+        channel.shutdown();
+      }
+    }
+    startServer(serverFlowControlWindow);
+
+    int channelPort = port;
+    if (useProxy) {
+      channelPort = proxyPort;
+    }
+    channel = NettyChannelBuilder.forAddress(new InetSocketAddress(loopback, channelPort))
+        .flowControlWindow(clientFlowControlWindow).negotiationType(NegotiationType.PLAINTEXT)
+        .build();
+  }
+
+  private void startServer(int serverFlowControlWindow) {
+    ServerBuilder<?> builder = NettyServerBuilder.forAddress(new InetSocketAddress(loopback, port))
+        .flowControlWindow(serverFlowControlWindow);
+    List<ServerInterceptor> allInterceptors = ImmutableList.<ServerInterceptor>builder()
+        .build();
+    builder.addService(ServerInterceptors.intercept(
+        TestServiceGrpc.bindService(new TestServiceImpl(Executors.newScheduledThreadPool(2))),
+        allInterceptors));
+    try {
+      server = builder.build().start();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Restart proxy with new bandwidth/latency settings.
+   *
+   * @param targetBPS
+   * @param targetLatency
+   * @param latencyUnits
+   */
+  private void resetProxy(final int targetBPS, final int targetLatency,
+      final TimeUnit latencyUnits) {
+    executor.submit(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          proxy.reset(targetBPS, targetLatency, latencyUnits);
+        } catch (Exception e) {
+          throw new RuntimeException(e);
+        }
+      }
+    });
+    try {
+      // give the a few millis to start before trying to re-connect. Without this, the first request
+      // sometimes throws an HTTP/2 error for a bad message.
+      Thread.sleep(10);
+    } catch (InterruptedException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private static void startProxy() {
+    executor.submit(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          proxy.start();
+        } catch (Exception e) {
+          throw new RuntimeException(e);
+        }
+      }
+    });
+  }
+
+
+  /**
+   * Simple stream observer to measure elapsed time of the call.
+   */
+  private class TestStreamObserver implements StreamObserver<StreamingOutputCallResponse> {
+
+    long lastRequest;
+    long elapsedTime;
+    CountDownLatch latch = new CountDownLatch(1);
+
+    public TestStreamObserver() {
+      lastRequest = System.nanoTime();
+    }
+
+    @Override
+    public void onNext(StreamingOutputCallResponse value) {}
+
+    @Override
+    public void onError(Throwable t) {
+      t.printStackTrace();
```

Comment:
There needs to be a better way of propagating this error.  What could cause this to happen?  How will you make sure the test fails if this happens?